starlette/staticfiles.py

```diff
@@ -109,9 +105,30 @@ async def get_response(self, path: str, scope: Scope) -> Response:
         Returns an HTTP response, given the incoming path, method and request headers.
         """
         if scope["method"] not in ("GET", "HEAD"):
-            return PlainTextResponse("Method Not Allowed", status_code=405)
+            raise HTTPException(status_code=405)
 
-        full_path, stat_result = await self.lookup_path(path)
+        try:
+            full_path, stat_result = await anyio.to_thread.run_sync(
+                self.lookup_path, path
+            )
+        except (FileNotFoundError, NotADirectoryError):
+            if self.html:
+                # Check for '404.html' if we're in HTML mode.
+                full_path, stat_result = await anyio.to_thread.run_sync(
+                    self.lookup_path, "404.html"
+                )
+                if stat_result is not None and stat.S_ISREG(stat_result.st_mode):
+                    return FileResponse(
+                        full_path,
+                        stat_result=stat_result,
+                        method=scope["method"],
+                        status_code=404,
+                    )
```

Comment:
Why are we not using `self.file_response` here?

Reply:
Makes sense, changed that.

Reply:
This was actually a question, because the `self.file_response` does more than that: 
![image](https://user-images.githubusercontent.com/7353520/135875059-d7524d0d-ae31-47ff-a127-2bb4ae56c32f.png)

Reply:
check the `self.is_not_modified` logic. It's the reason why ur tests are failing.

Reply:
Good catch, I didn't remember that change :D
Now that I take a look, looks like the answer to the question is "No". We can't use `self.file_response` as it will cache `404` responses. We have to create `FileResponse` there.

Reply:
People usually assume that I'm suggesting things, when I'm innocently asking questions hahaha

Thanks!